row-data-gateway/README.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: pattern
+title: Row Data Gateway
+folder: row-data-gateway
+permalink: /patterns/row-data-gateway/
+categories: Structural
+tags:
+ - Data access
+---
+## Intent
+An object that acts as a Gateway to a single record in a data source. There is one instance per row. Here Gateway means an object that encapsulates access to an external system or resource. This object does not contain domain logic methods. If you introduce other methods (in particular domain logic) the object becomes an Active Record Pattern.
+
+## Explanation
+
+In plain words
+
+> A Row Data Gateway gives you objects that look exactly like the record in your record structure but can be accessed with the regular mechanisms of your programming language. All details of data source access are hidden behind this interface.
+
+Programmatic Example
+
+A Row Data Gateway acts as an object that exactly mimics a single record, such as one database row. For example, `Person` class has id, firstName and lastName fields. 
+
+The `Person` class:
+
+```java
+@Setter
+@Getter
+@ToString
+@EqualsAndHashCode
+@AllArgsConstructor
+public class Person {
+  private int id;
+  private String firstName;
+  private String lastName;
+}
+```
+
+This pattern holds the data about a row so that a client can then access the Row Data Gateway directly. The gateway acts as a good interface for each row of data. This approach works particularly well for Transaction Scripts.
+
+The `PersonGateway` class:
+
+```java
+public class PersonGateway {
+  private final DataSource dataSource;
+  private final Person person;
+
+  public PersonGateway(final Person person1, final DataSource dataSource1) {
+    this.person = person1;
+    this.dataSource = dataSource1;
+  }
+
+  public int insert() throws SQLException {
+	...
+  }
+  
+  public int update() throws SQLException {
+	...
+  }
+
+  public int delete() throws SQLException {
+ 	...
+  }
+}
+```

```

Comment:
Continue the example by showing how to insert, update, delete records. Finally, show console output.